### DISCUSSION TITLE
Batch Requests for upsert function import - SAP SuccessFactors

### DISCUSSION THREAD

**Discussion by vdevExplorer** [SAP/cloud-sdk-js#4122] [Q&A]
Hi, looking some guidance on how to implement batch request for upsert function import  in SAP Successfactors. We have a use case where we need to upsert multiple EC entities in a single transaction. I tried using cloud sdk odata client to generate but no luck in calling the batch.

Thanks,

**Comment by deekshas8** [comment]
Hi @vinay-devalla ,

You can use the `changeset()` function to bundle all your requests and then pass it to the `batch()`. The [changeset](https://sap.github.io/cloud-sdk/docs/js/features/odata/v2-client#changeset) documentation provides an example.

**Reply by vdevExplorer** [reply]
Hi @deekshas8  - Thanks for following up. The issue I am facing is the batch is failing with ""lang" : "en-US", "value" : "ChangeSet index 1 - Upsert content should NOT be empty"\n'" . I couldnt pass the payload content to the function import.

**Reply by deekshas8** [reply]
Hi @vdevExplorer ,
Apologies for the delayed response here. Could you share the EDMX file of the service and let me know which function import you are trying to invoke and also share the code snippet. I can then take a look if this is something the SDK supports.
